### PR TITLE
Initial pass at strimzi sample using the spec

### PR DESCRIPTION
**Pull request by arthurdm**
Moving the strimzi sample off the main `spec` repo.   Converted to use the spec's `ServiceBinding` CR.   Things aren't actually working because SBO needs to implement parts of the spec, but at least it's more representative of what the sample should look like.

Signed-off-by: Arthur De Magalhaes <ademagalhaes@gmail.com>

### CODE REVIEW

File: strimzi/04-service-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    kind: RuntimeComponent
+    name: kafka-producer
+  service:
+    apiVersion: v1beta1/kafka.strimzi.io
+    kind: Kafka
+    name: my-producer-cluster
+    mappings:
+      - name: hosts
+        value: |-
+          {{- range .status.listeners -}}
```
Comment (scothis) [inline strimzi/04-service-binding.yaml:31]: so happy we didn't define a custom grammar

File: strimzi/04-service-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+kind: ServiceBinding
+metadata:
+  name: kafka-producer-binding-user
+spec:
+  application:
+    apiVersion: app.stacks/v1beta1
+    kind: RuntimeComponent
+    name: kafka-producer
+  service:
+    apiVersion: v1beta1/kafka.strimzi.io
```
Comment (scothis) [inline strimzi/04-service-binding.yaml]:
```suggestion
    apiVersion: kafka.strimzi.io/v1beta1
```

File: strimzi/04-service-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+kind: ServiceBinding
+metadata:
+  name: kafka-producer-binding-cluster
+spec:
+  application:
+    apiVersion: app.stacks/v1beta1
+    kind: RuntimeComponent
+    name: kafka-producer
+  service:
+    apiVersion: v1beta1/kafka.strimzi.io
```
Comment (scothis) [inline strimzi/04-service-binding.yaml]:
```suggestion
    apiVersion: kafka.strimzi.io/v1beta1
```

File: strimzi/04-service-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  kind: ServiceBinding
+  metadata:
+    name: kafka-consumer-binding-user
+  spec:
+    application:
+      apiVersion: app.stacks/v1beta1
+      kind: RuntimeComponent
+      name: kafka-consumer
+    service:
+      apiVersion: v1beta1/kafka.strimzi.io
```
Comment (scothis) [inline strimzi/04-service-binding.yaml]:
```suggestion
      apiVersion: kafka.strimzi.io/v1beta1
```

File: strimzi/04-service-binding.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  kind: ServiceBinding
+  metadata:
+    name: kafka-consumer-binding-cluster
+  spec:
+    application:
+      apiVersion: app.stacks/v1beta1
+      kind: RuntimeComponent
+      name: kafka-consumer
+    service:
+      apiVersion: v1beta1/kafka.strimzi.io
```
Comment (scothis) [inline strimzi/04-service-binding.yaml]:
```suggestion
      apiVersion: kafka.strimzi.io/v1beta1
```

File: strimzi/README.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+The cluster admin needs to install 3 operators into the cluster:
+
+* Service Binding Operator
+* Strimzi Operator
+* Runtime Component Operator
+
+#### Install the Service Binding Operator
+
+Navigate to the **Operator** → **OperatorHub** in the OpenShift console. From the `Developer Tools` category, select the `Service Binding Operator` operator and install a `alpha` version. This would install the `ServiceBindingRequest` *custom resource definition* (CRD) in the cluster.
```
Comment (scothis) [inline strimzi/README.md:29]: Would be nice to include instructions that work outside of OpenShift as well

File: strimzi/04-service-binding.yaml (hunk 1 above)
Comment (arthurdm) [reply strimzi/04-service-binding.yaml:31]: definitely!

File: strimzi/04-service-binding.yaml (hunk 2 above)
Comment (arthurdm) [reply strimzi/04-service-binding.yaml]: good catch.  doing PRs during meetings is never a good idea.  😅

File: strimzi/README.md (hunk 6 above)
Comment (arthurdm) [reply strimzi/README.md:29]: yup, I agree.  I will open up an issue to follow up on this sample, and have instructions for vanilla k8s too.

Comment (arthurdm) [reply strimzi/README.md:29]: ref #2

File: strimzi/01-kafka.yaml [hunk 7]
```diff
@@ -0,0 +1,64 @@
+apiVersion: kafka.strimzi.io/v1alpha1
```
Comment (navidsh) [inline strimzi/01-kafka.yaml]:
```suggestion
apiVersion: kafka.strimzi.io/v1beta1
```

File: strimzi/04-service-binding.yaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  name: kafka-producer-binding-cluster
+spec:
+  application:
+    apiVersion: app.stacks/v1beta1
+    kind: RuntimeComponent
+    name: kafka-producer
+  service:
+    apiVersion: kafka.strimzi.io/v1beta1
+    kind: Kafka
+    name: my-producer-cluster
```
Comment (navidsh) [inline strimzi/04-service-binding.yaml]: I think both the producer and the consumer will be exchanging messages with the same Kafka Cluster, `my-cluster`:
```suggestion
    name: my-cluster
```

File: strimzi/04-service-binding.yaml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    name: kafka-consumer-binding-cluster
+  spec:
+    application:
+      apiVersion: app.stacks/v1beta1
+      kind: RuntimeComponent
+      name: kafka-consumer
+    service:
+      apiVersion: kafka.strimzi.io/v1beta1
+      kind: Kafka
+      name: my-consumer-cluster
```
Comment (navidsh) [inline strimzi/04-service-binding.yaml]: I think both the producer and the consumer will be exchanging messages with the same Kafka Cluster, `my-cluster`:
```suggestion
    name: my-cluster
```

**Review by navidsh — COMMENTED** [review]
Minor comments

File: strimzi/04-service-binding.yaml (hunk 9 above)
Comment (arthurdm) [reply strimzi/04-service-binding.yaml]: good catch!

**Review by nebhale — APPROVED** [review]

**Pull request merged**